### PR TITLE
ncdu: move to native tcell

### PR DESCRIPTION
**Pull request by eNV25**
#### What is the purpose of this change?

Based the commit on #6562.

Use tcell directly instead of the termbox wrapper

Following up on 36add0af, which switched from `termbox`
to `tcell`'s wrapper.

Here are some differences from the previous code.

##### Graphemes are taken into account

The original `termbox` library only supports using one rune (code-point) for a cell. But many Unicode characters are made up of multiple runes (code-points), called grapheme clusters.

With the `tcell` library, you can also set additional runes, which seems to work for graphemes. This code iterates through graphemes through the `uniseg` library, which seems to be an indirect dependency already.

#### Was the change discussed in an issue or in the forum before?

Previous: #6258

Issue:  #6272

#### Checklist

- [x] I have read the [contribution guidelines](https://github.com/rclone/rclone/blob/master/CONTRIBUTING.md#submitting-a-new-feature-or-bug-fix).
- [x] I have added tests for all changes in this PR if appropriate.
- [x] I have added documentation for the changes if appropriate.
- [x] All commit messages are in [house style](https://github.com/rclone/rclone/blob/master/CONTRIBUTING.md#commit-messages).
- [x] I'm done, this Pull Request is ready for review :-)

### CODE REVIEW

File: cmd/ncdu/ncdu.go [hunk 1]
```diff
@@ -150,77 +152,91 @@ type dirPos struct {
 	offset int
 }
 
+// graphemeWidth returns the number of cells in rs.
+//
+// The original [runewidth.StringWidth] iterates through graphemes
+// and uses this same logic. To avoid iterating through graphemes
+// repeatedly, we separate that out into its own function.
+func graphemeWidth(rs []rune) (wd int) {
+	// copied/adapted from [runewidth.StringWidth]
```
Comment (eNV25) [inline cmd/ncdu/ncdu.go:161]: This is something I'm not sure about. Do I properly attribute this code fragment?

The original code is MIT licensed.

Comment (eNV25) [reply cmd/ncdu/ncdu.go:161]: @ncw

File: cmd/ncdu/ncdu.go [hunk 2]
```diff
@@ -926,39 +933,50 @@ outer:
...
+				if k := ev.Key(); k == tcell.KeyRune {
+					c = ev.Rune()
+				} else {
+					c = key(k)
+				}
+				switch c {
+				case key(tcell.KeyEsc), key(tcell.KeyCtrlC), 'q':
 					if u.showBox {
-						u.showBox = false
+						u.showBox = !u.showBox
```
Comment (ncw) [inline cmd/ncdu/ncdu.go:955]: This appears to be a change in operation?

File: cmd/ncdu/ncdu.go (hunk 1 above)
Comment (ncw) [reply cmd/ncdu/ncdu.go:161]: I think this is a small enough bit of code that no-one is going to be bothered

**Review by ncw — COMMENTED** [review]
Thanks for doing this work :-)

How important is the grapheme support? Are there any terminals which actually support graphemes? The ones I tested didn't.

I put a couple of comments inline also.

File: cmd/ncdu/ncdu.go (hunk 2 above)
Comment (eNV25) [reply cmd/ncdu/ncdu.go]: True. I had fixed it, but I forgot to push.

**Comment by eNV25** [comment]
> How important is the grapheme support? Are there any terminals which actually support graphemes? The ones I tested didn't.

The previous version doesn't work for accented characters made up of two code points. e.g. `A + ◌́ => Á`. This work correctly in this PR.

```
$ touch 'Á'
$ go run . ncdu .
```

But this PR isn't perfect either (take for example this emoji `🐕‍🦺`), but I think it works well enough. I would have to ask tcell's author on how to best handle this.

File: cmd/ncdu/ncdu.go [hunk 3]
```diff
@@ -1001,15 +1022,18 @@ outer:
...
 		}
 		// listen to key presses, etc.
 	}
 	return nil
 }
+
+// key returns a rune representing the key k. It is larger than the maximum Unicode code-point.
+func key(k tcell.Key) rune {
+	// This is the maximum possible Unicode code point. Anything greater fails to compile as a Go quoted rune literal.
+	return '\U0010FFFF' + rune(k)
```
[the remark this answers is not in the document]
Comment (ncw) [reply cmd/ncdu/ncdu.go:1038]: I think it is OK as an innovative solution :-)

**Review by ncw — APPROVED** [review]
I think this is ready to merge now, so I will do so now - thank you :-)

**Pull request merged**